lms/static/scripts/frontend_apps/components/test/LMSGrader-test.js

```diff
@@ -29,15 +32,19 @@ describe('LMSGrader', () => {
 
   beforeEach(() => {
     $imports.$mock({
-      '../utils/update-client-config': {
-        updateClientConfig: fakeUpdateClientConfig,
-        removeClientConfig: fakeRemoveClientConfig,
-      },
       './StudentSelector': FakeStudentSelector,
+      '../../postmessage_json_rpc/client': {
+        call: fakeRpcCall,
+      },
+      '../../postmessage_json_rpc/server': {
+        getSidebarWindow: fakeSidebarWindow,
+      },
     });
   });
 
   afterEach(() => {
+    fakeRpcCall.resetHistory();
+    fakeOnChange.resetHistory();
```

Comment:
One reason we usually create new stubs in `beforeEach` rather than having a single shared stub that is reset between tests is that this creates a hazard where a test can trigger some activity (eg. in a timeout or Promise) that ends up affecting a later test due to sharing the same state. If each test gets a fresh stub instance, this can't happen.

Reply:
yes, i think thats the issue I ran into, so I'll go ahead and make sure each stub is brand new as that seems to be a much better standard. I sorta think things like `resetHistory` lead to less than ideal patterns and perhaps that function should just not be part of sinon. The API is quiet huge and hard to lock in best practices with since there are so many ways to do things.